awsbestpractices/s3-object-user-metadata-key-case-sensitivity/S3ObjectUserMetadataKeyCaseSensitivity.java

```diff
@@ -0,0 +1,83 @@
+package example;
+
+public class Example {
+    
+    ObjectMetadata objMetaData;
+
+    void uppercaseAtCall1() {
+        objMetaData.getUserMetaDataOf("Key");
+    }
+
+    void lowercaseAtCall1() {
+        objMetaData.getUserMetaDataOf("key");
+    }
+
+    void uppercaseAtCall2() {
+        objMetaData.getUserMetadata().get("Key");
+    }
+
+    void lowercaseAtCall2() {
+        objMetaData.getUserMetadata().get("key");
+    }
+
+    void uppercaseFromVariable1() {
+        String var = "Key";
+        objMetaData.getUserMetaDataOf(var);
+    }
+
+    void lowercaseFromVariable1() {
+        String var = "key";
+        objMetaData.getUserMetaDataOf(var);
+    }
+
+    void uppercaseFromVariable2() {
+        String var = "Key";
+        objMetaData.getUserMetadata().get(var);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The user-metadata key used in method call get() on this line contains an uppercase character from the string &#34;Key&#34; on line 34. It has to be a lowercase string [as specified in the documentation](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/ObjectMetadata.html#getUserMetadata--), even if it was originally an uppercase string.